### ISSUE TITLE
Binary blobs in `hal_telink` submodules

### ISSUE THREAD

**Issue by carlescufi** [zephyrproject-rtos/zephyr#52113]
[This PR](https://github.com/zephyrproject-rtos/zephyr/pull/45903) added support for Bluetooth in Telink platforms. However, this also [updated the manifest](https://github.com/zephyrproject-rtos/zephyr/blob/90577ee5d8b5ac6eea891134e16aad7b69e239dc/west.yml#L131-L134) with a revision that includes a [submodule with precompiled libraries](https://github.com/zephyrproject-rtos/hal_telink/pull/4/files#diff-fe7afb5c9c916e521401d3fcfb4277d5071798c3baf83baf11d6071742823584).

The `hal_telink` tree thus [now contains a submodule](https://github.com/zephyrproject-rtos/hal_telink/tree/main/tlsr9/ble/proj_lib) which includes [two libraries](https://github.com/telink-semi/zephyr_hal_telink_b91_ble_lib/tree/0a49c9bf3cfe4684cc27fbaa6ad070186703e452).

This submodule needs to be removed and the libraries need to use the new [binary blob infrastructure](https://docs.zephyrproject.org/latest/contribute/bin_blobs.html).

**Comment by tejlmand** [comment]
Fixed by: #52112 
but maybe @yurvyn would like to post a PR which adopts to the approved bin blobs scheme:
https://docs.zephyrproject.org/latest/contribute/bin_blobs.html

**Comment by thedjnK** [comment]
@interfer

**Comment by jhedberg** [comment]
> Fixed by: #52112 but maybe @yurvyn would like to post a PR which adopts to the approved bin blobs scheme: https://docs.zephyrproject.org/latest/contribute/bin_blobs.html

Seems like this never happened, so the Zephyr Bluetooth driver for it has been unbuildable for several years now (without having to resort to non-standard hacks). Seems to me like it should probably just be removed from Zephyr.

**Comment by jhedberg** [comment]
> > Fixed by: #52112 but maybe @yurvyn would like to post a PR which adopts to the approved bin blobs scheme: https://docs.zephyrproject.org/latest/contribute/bin_blobs.html
> 
> Seems like this never happened, so the Zephyr Bluetooth driver for it has been unbuildable for several years now (without having to resort to non-standard hacks). Seems to me like it should probably just be removed from Zephyr.

Additionally, the code seems to have bitrotten to the point where it doesn't build anymore. I added `submodules: true` back to `west.yml` and tried to build `samples/bluetooth/peripheral`, with the following result:
```
[117/190] Building C object modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/drivers/B91/flash.c.obj
FAILED: modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/drivers/B91/flash.c.obj
ccache /Users/jhedberg/bin/zephyr-sdk-0.16.5/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=4096 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/Users/jhedberg/src/zephyr/zephyr/include -I/Users/jhedberg/src/zephyr/zephyr/build/zephyr/include/generated -I/Users/jhedberg/src/zephyr/zephyr/soc/telink/tlsr -I/Users/jhedberg/src/zephyr/zephyr/soc/common/riscv-privileged/. -I/Users/jhedberg/src/zephyr/zephyr/soc/telink/tlsr/tlsr951x/. -I/Users/jhedberg/src/zephyr/zephyr/subsys/bluetooth -I/Users/jhedberg/src/zephyr/zephyr/subsys/settings/include -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/common -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91/compatibility_pack -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91/reg_include -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91 -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/vendor/controller -I/Users/jhedberg/src/zephyr/modules/crypto/tinycrypt/lib/include -isystem /Users/jhedberg/src/zephyr/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /Users/jhedberg/src/zephyr/zephyr/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mabi=ilp32 -march=rv32imac_zicsr_zifencei --sysroot=/Users/jhedberg/bin/zephyr-sdk-0.16.5/riscv64-zephyr-elf/riscv64-zephyr-elf -imacros /Users/jhedberg/src/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/Users/jhedberg/src/zephyr/zephyr/samples/bluetooth/peripheral=CMAKE_SOURCE_DIR -fmacro-prefix-map=/Users/jhedberg/src/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/Users/jhedberg/src/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/drivers/B91/flash.c.obj -MF modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/drivers/B91/flash.c.obj.d -o modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/drivers/B91/flash.c.obj -c /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91/flash.c
In file included from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ble_controller.h:32,
                 from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91/flash.c:36:
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/ll.h:118:1: error: unknown type name 'bool'
  118 | bool            blc_ll_isValidPublicAddr(u8* addr);
      | ^~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/ll.h:1:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
  +++ |+#include <stdbool.h>
    1 | /******************************************************************************
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/ll.h:126:1: error: unknown type name 'bool'
  126 | bool            blc_ll_isValidRandomAddr(u8* addr);
      | ^~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/ll.h:126:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/ll.h:135:1: error: unknown type name 'bool'
  135 | bool            blc_ll_isValidOwnAddrByAddrType(u8 ownAddrType, u8* randomAddr);
      | ^~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/ll.h:135:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
In file included from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ble_controller.h:35:
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/acl_conn/acl_conn.h:152:1: error: unknown type name 'bool'
  152 | bool            blc_ll_isAclConnEstablished(u16 connHandle);
      | ^~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ll/acl_conn/acl_conn.h:1:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
  +++ |+#include <stdbool.h>
    1 | /******************************************************************************
In file included from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ble_controller.h:77:
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/phy/phy_test.h:71:1: error: unknown type name 'bool'
   71 | bool      blc_phy_isPhyTestEnable(void);
      | ^~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/phy/phy_test.h:1:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
  +++ |+#include <stdbool.h>
    1 | /******************************************************************************
In file included from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/algorithm/algorithm.h:26,
                 from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/stack/ble/controller/ble_controller.h:79:
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/algorithm/ecc/ecc_ll.h:45:106: error: unknown type name 'bool'
   45 | int                     blt_ecc_gen_key_pair(unsigned char *pub, unsigned char *priv, ecc_curve_t curve, bool use_dbg_key);
      |                                                                                                          ^~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/algorithm/ecc/ecc_ll.h:22:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
   21 | #include "algorithm/ecc/ecc_curve.h"
  +++ |+#include <stdbool.h>
   22 |
[120/190] Building C object modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/ble/vendor/controller/b91_bt.c.obj
FAILED: modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/ble/vendor/controller/b91_bt.c.obj
ccache /Users/jhedberg/bin/zephyr-sdk-0.16.5/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=4096 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/Users/jhedberg/src/zephyr/zephyr/include -I/Users/jhedberg/src/zephyr/zephyr/build/zephyr/include/generated -I/Users/jhedberg/src/zephyr/zephyr/soc/telink/tlsr -I/Users/jhedberg/src/zephyr/zephyr/soc/common/riscv-privileged/. -I/Users/jhedberg/src/zephyr/zephyr/soc/telink/tlsr/tlsr951x/. -I/Users/jhedberg/src/zephyr/zephyr/subsys/bluetooth -I/Users/jhedberg/src/zephyr/zephyr/subsys/settings/include -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/common -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91/compatibility_pack -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91/reg_include -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/drivers/B91 -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble -I/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/vendor/controller -I/Users/jhedberg/src/zephyr/modules/crypto/tinycrypt/lib/include -isystem /Users/jhedberg/src/zephyr/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /Users/jhedberg/src/zephyr/zephyr/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mabi=ilp32 -march=rv32imac_zicsr_zifencei --sysroot=/Users/jhedberg/bin/zephyr-sdk-0.16.5/riscv64-zephyr-elf/riscv64-zephyr-elf -imacros /Users/jhedberg/src/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/Users/jhedberg/src/zephyr/zephyr/samples/bluetooth/peripheral=CMAKE_SOURCE_DIR -fmacro-prefix-map=/Users/jhedberg/src/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/Users/jhedberg/src/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/ble/vendor/controller/b91_bt.c.obj -MF modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/ble/vendor/controller/b91_bt.c.obj.d -o modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/ble/vendor/controller/b91_bt.c.obj -c /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/vendor/controller/b91_bt.c
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/vendor/controller/b91_bt.c:19:10: fatal error: zephyr.h: No such file or directory
   19 | #include <zephyr.h>
      |          ^~~~~~~~~~
compilation terminated.
[124/190] Building C object modules/hal_telink/tlsr9/CMakeFiles/..__modules__hal__telink__tlsr9.dir/ble/common/utility.c.obj
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/common/utility.c:53:16: warning: argument 1 of type 'u8[7]' {aka 'unsigned char[7]'} with mismatched bound [-Warray-parameter=]
   53 | void swap48(u8 dst[7], const u8 src[7])
      |             ~~~^~~~~~
In file included from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/tl_common.h:29,
                 from /Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/common/utility.c:18:
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/common/utility.h:123:16: note: previously declared as 'u8[6]' {aka 'unsigned char[6]'}
  123 | void swap48(u8 dst[6], const u8 src[6]);
      |             ~~~^~~~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/common/utility.c:53:33: warning: argument 2 of type 'const u8[7]' {aka 'const unsigned char[7]'} with mismatched bound [-Warray-parameter=]
   53 | void swap48(u8 dst[7], const u8 src[7])
      |                        ~~~~~~~~~^~~~~~
/Users/jhedberg/src/zephyr/modules/hal/telink/tlsr9/ble/common/utility.h:123:33: note: previously declared as 'const u8[6]' {aka 'const unsigned char[6]'}
  123 | void swap48(u8 dst[6], const u8 src[6]);
      |                        ~~~~~~~~~^~~~~~
[130/190] Building C object zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/hci_core.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /opt/homebrew/bin/cmake --build /Users/jhedberg/src/zephyr/zephyr/build
```

At this point it looks like this just isn't maintained anymore and would be a good candidate for removal from Zephyr.

**Comment by andy-liu-telink** [comment]
@jhedberg 

@yurvyn is NO longer with our team.  Other team members will look into this issue.

@s07641069
@andriy-bilynskyy
Could you take a look?
